### ISSUE TITLE
Disabled Buttons

### ISSUE THREAD

**Issue by SpencerReynolds18** [carbon-design-system/carbon#584]
## Detailed description

Disabled Buttons are needed but present challenges. Disabled items are not read by screenreaders making disabled buttons inaccessible. Additionally, we cannot place tooltips on disabled buttons. This makes it difficult to provide users additional information about why the modal has been disabled. 

> Is this issue related to a specific component? 
Disabled Buttons

> What did you expect to happen? What happened instead? What would you like to see changed?
I expected to be able to place a tooltip on a disabled button that was accessible by screenreaders and normal users.

## Example use case
Here is where we tried to add tooltips to disabled buttons in order to inform users that they could not take certain actions until they updated their version of Kubernetes. We ran into the roadblocks above and our UI squad came up with this workaround where they triggered the tooltip off of the tooltip icon within the button. It is a little bit of a hack(and unreadable by screenreaders) but the most elegant solution we could do with the existing components. 

![gif_tooltips_disabled_buttons](https://user-images.githubusercontent.com/19826147/36322659-cf1c23f4-1313-11e8-970b-302df7d8fe35.gif)

## Potential Solution
What if the button was left as active, but had "disabled" styling applied? We could also somehow void the javascript or action that the button normally would execute. Because the button was still active, a tooltip could be applied and screenreaders could still navigate to and hopefully read the tooltip info about why the button was disabled in function and styling(but still an active button by coding standards).

**Comment by SpencerReynolds18** [comment]
Had someone reach out asking about this issue as they have a similar need (they were unable to comment):
<img width="707" alt="screen shot 2018-02-26 at 10 42 08 am" src="https://user-images.githubusercontent.com/19826147/36682853-c8c7012e-1ae1-11e8-85e6-20e30d7ae7f5.png">

**Comment by ceci-work** [comment]
Hi! we are going to have this issue a lot in our UI. Account owners will be giving certain users reader roles to buckets but not writer roles which means that when they go to the UI many CTA's will not be actionable, I feel we need a systematic solution on how to manage these limited roles on all UI items to limit: visibility and editability.

**Comment by SpencerReynolds18** [comment]
Just checking in on this one again. It came up again in a different area of our service.
